### DISCUSSION TITLE
ModuleNotFoundError with openpyxl

### DISCUSSION THREAD

**Discussion by dharmikP17** [pyinstaller/pyinstaller#8755] [Help]
I am trying to crate an exe file for GUI application which uses several modules, that are not found by the pyinstaller. These are openpyxl, win32api, win32print and so on. 

I am using pycharm for python and using virtual environment, does this affect somehow ?

After searching a bit I see there are solutions like using --hidden-import and --add-data however these did not work for me.

Below is the list of errors I have in text doc
```
missing module named pwd - imported by posixpath (delayed, conditional, optional), shutil (delayed, optional), tarfile (optional), pathlib (delayed, optional), subprocess (delayed, conditional, optional)
missing module named grp - imported by shutil (delayed, optional), tarfile (optional), pathlib (delayed, optional), subprocess (delayed, conditional, optional)
missing module named posix - imported by os (conditional, optional), posixpath (optional), shutil (conditional), importlib._bootstrap_external (conditional)
missing module named resource - imported by posix (top-level)
missing module named _frozen_importlib_external - imported by importlib._bootstrap (delayed), importlib (optional), importlib.abc (optional)
excluded module named _frozen_importlib - imported by importlib (optional), importlib.abc (optional)
missing module named img2pdf - imported by C:\Users\Tech01\Desktop\software\database\NM_Inventory_2.3.py (top-level)
missing module named win32print - imported by C:\Users\Tech01\Desktop\software\database\NM_Inventory_2.3.py (top-level)
missing module named win32api - imported by C:\Users\Tech01\Desktop\software\database\NM_Inventory_2.3.py (top-level)
missing module named _posixsubprocess - imported by subprocess (conditional)
missing module named fcntl - imported by subprocess (optional)
missing module named openpyxl - imported by C:\Users\Tech01\Desktop\software\database\NM_Inventory_2.3.py (top-level)
```

**Comment by rokm** [comment]
> I am using pycharm for python and using virtual environment, does this affect somehow ?

You need to have PyInstaller installed in the same virtual environment, and if you have PyInstaller installed in multiple environments (like in both base python environment and in virtual environment), you need to ensure that you are actually running PyInstaller that is installed within the target environment. To be sure, you can run `where pyinstaller` to see which `pyinstaller.exe` shim(s) you have in path; if there are multiple, you should try running `python -m PyInstaller ...` instead of `pyinstaller ...` after ensuring that your virtual environment is active.

**Reply by dharmikP17** [reply]
Yes, you were right. I had multiple pyinstaller installed and one I was using was not even in venv indeed. I just uninstalled all of them and started everything again in venv from Pycharm cmd.  This worked without any problem and exe file is also working. Thank you very much for the help.
